common/unixTime.go

```diff
@@ -0,0 +1,121 @@
+package common
+
+import (
+	"time"
+
+	"github.com/multiversx/mx-chain-go/errors"
+)
+
+const (
+	// NumberOfSecondsInDay defines the number of seconds in a day
+	NumberOfSecondsInDay = 86400
+	// NumberOfMillisecondsInDay defines the number of milliseconds in a day
+	NumberOfMillisecondsInDay = NumberOfSecondsInDay * 1000
+)
+
+const (
+	minRoundDurationMS  = 500
+	minRoundDurationSec = 1
+)
+
+// TimeToUnix returns the time to unix based on current configuration
+func TimeToUnix(
+	t time.Time,
+	enableEpochsHandler EnableEpochsHandler,
+) int64 {
+	if enableEpochsHandler.IsFlagEnabled(SupernovaFlag) {
+		return t.UnixMilli()
+	}
+
+	return t.Unix()
+}
+
+// TimeToUnixInEpoch returns the time to unix based on current configuration
+func TimeToUnixInEpoch(
```

Comment:
this as well not very clear what it returns, sometimes it returns milliseconds, other times it returns seconds.

Reply:
added a TODO here, will think a bit more on this, it might be better to avoid these functions, since they are not used in so many places and it might be safes to handle each case in place specifically

Reply:
and add some functions for handling genesis timestamp (since this is used in multiple places in factories)